backend/src/users/services/users.service.ts

```diff
@@ -0,0 +1,78 @@
+import { ConflictException, ForbiddenException, Injectable, NotFoundException } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { User } from "users/model/users.entity";
+import { UserCredentialsDto } from "../model/dto/userCredentials.dto";
+import { UpdateUserDto } from "../model/dto/updateUser.dto";
+
+@Injectable()
+export class UsersService {
+  constructor(
+    @InjectRepository(User) private userRepository: Repository<User>
+  ) {
+  }
+
+  async create(userCredentialsDto: UserCredentialsDto) {
+    const user = await this.userRepository.findOneBy({
+      username: userCredentialsDto.username
+    });
+
+    if (user) throw new ConflictException("Duplicated username");
+
+    return this.userRepository.save(userCredentialsDto);
+  }
+
+  async update(userId: number, paramId: number, updateUserDto: UpdateUserDto) {
+    updateUserDto.username = updateUserDto.username
+      .replace(/\s/g, "").toLowerCase();
+
+    if (userId !== paramId) throw new ForbiddenException();
+
+    const { username } = await this.findOneById(paramId, userId);
+
+    if (username !== updateUserDto.username) {
+      const { affected } = await this.userRepository.update(paramId, updateUserDto).catch(() => {
+        throw new ConflictException("Duplicated username");
+      });
+
+      return Boolean(affected);
```

Comment:
Don't mix await with then/catch. Use try-catch with await instead.

Reply:
After conversation, leave it like that

Reply:
After conversation, leave it like that